### DISCUSSION TITLE
Large Benchmark Discrepancy NRX vs WEG on RX 6900 XT LC

### DISCUSSION THREAD

**Discussion by cmarhoover** [ChefKissInc/NootRX#59] [Q&A] [answered]
Hi all, thanks for the extensive effort to make RDNA2 even easier to use in hacks.

I have a somewhat uncommon Navi 21 variant - the 6900 XT LC [as shown here](https://www.techpowerup.com/gpu-specs/amd-radeon-rx-6900-xt-lc.b9071). It's an XTXH with a modest overclock and memory bandwidth increase. I've been running with no trouble in my hack by spoofing with WhateverGreen (currently version 1.6.6)  for a few years now without issue. After upgrading to macOS 14, I can even take full advantage of my 4K 144Hz VRR HDR monitor. Super.

Out of curiosity I tried the [latest build](https://github.com/ChefKissInc/NootRX/commit/b967a1362ea7c769441d81a5244bc89e384d570e) of NootRX. I stripped out all the WEG boot args and spoofing-related items from my config, and was able to boot with no problems. The desktop and apps felt snappy as usual. However, there is a major disparity in Geekbench 6 Compute scores when switching from one kext to the other that has me scratching my head. 

Despite this not really being a problem - my system is perfectly usable in any case - I'm wondering if anyone can shed some light on what could be going on with the benchmark performance. Is it just the relative maturity of NRX and WEG? Something about this RX 6900 variant? Or just some boneheaded mistake I made while switching kexts?

Attached screenshots compare benchmark runs between NRX and WEG, both under macOS 14.5 and Opencore 1.0.0, with only NRX/WEG related changes to the config. All other system parameters are the same across benchmark runs as well..

- Ryzen 5950X, Per Core CurveOptimizer enabled
- 64GB DDR4 3200MHz 
- Aorus Elite X570, Above 4G Decoding enabled, Resizable BAR enabled 
- Triple Monitor, all 4K, 1 of 3 is high refresh rate HDR

Thanks! 

![2024-05-25_19-17](https://github.com/ChefKissInc/NootRX/assets/17608264/a47027a7-e242-48c9-9fef-c1e57a994e74)

![2024-05-25_19-42](https://github.com/ChefKissInc/NootRX/assets/17608264/ae47e350-cd81-4037-876a-7736aac8b680)
![2024-05-25_19-47](https://github.com/ChefKissInc/NootRX/assets/17608264/b025d9f2-6149-4a4f-b922-99508543ab9d)

Hackintool shows GFX0 like this under WEG 1.6.6:
![2024-05-25_19-39](https://github.com/ChefKissInc/NootRX/assets/17608264/5770f90f-16f6-4526-9e6b-3ec85a56fcf6)

...and like this under NRX [b967a13](https://github.com/ChefKissInc/NootRX/commit/b967a1362ea7c769441d81a5244bc89e384d570e)
![2024-05-25_19-57](https://github.com/ChefKissInc/NootRX/assets/17608264/d2639dce-58cc-477b-bad5-8ccd109e3022)

**Comment by VisualEhrmanntraut** [comment, marked as answer]
Hi, the developer of NootRX here. The major difference between WEG and my kext is that my kext updates the firmware and ASIC settings and switches the workload policy to a safer one (because apple sets it as VR which can cause coil whine and high temperatures/power waste while Windows uses Default and so does my kext). There might be some other things I have not taken note of. The firmware update is also not 100% completed because I need to make a better way to inject the firmware to do it properly.
